Parser/asdl_c.py

```diff
@@ -71,6 +71,21 @@ def is_simple(sum):
             return False
     return True
 
+def asdl_of(name, obj):
+    if isinstance(obj, asdl.Product) or isinstance(obj, asdl.Constructor):
+        fields = ", ".join(map(str, obj.fields))
+        if fields:
+            fields = "({})".format(fields)
+        return "{}{}".format(name, fields)
+    else:
+        if is_simple(obj):
+            types = " | ".join(type.name for type in obj.types)
+        else:
+            template = "\\n{}| ".format(" " * (len(name) + 3))
```

Comment:
It is rather not `template` but `sep`.

And why `\\n`? Should not it be `\n`?

And I would use `+ 1` instead of `+ 3`.
```suggestion
            sep = "\n{}| ".format(" " * (len(name) + 1))
```

Reply:
Thanks, `+1` is that it should be. I dont know why I choosed `+3`. Will replace `template` with `sep`.